core/drivers/crypto/caam/cipher/caam_cipher.c

```diff
@@ -0,0 +1,1052 @@
+		break;
+
+	case TEE_MAIN_ALGO_DES:
+		if (algo_md < ARRAY_SIZE(des_alg))
+			ca = &des_alg[algo_md];
+		break;
+
+	case TEE_MAIN_ALGO_DES3:
+		if (algo_md < ARRAY_SIZE(des3_alg))
+			ca = &des3_alg[algo_md];
+		break;
+
+	default:
+		break;
+	}
+
+	if (ca)
+		if (ca->type)
+			return ca;
+
+	return NULL;
+}
+
+/*
+ * Allocate the SW cipher data context
+ *
+ * @algo  Algorithm ID of the context
+ * @ctx   [out] Caller context variable
+ */
+static TEE_Result do_allocate(void **ctx, uint32_t algo)
+{
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+	struct cipherdata *cipherdata = NULL;
+	const struct cipheralg *alg = NULL;
+
+#ifdef CFG_CAAM_64BIT
+#define MAX_DESC_ENTRIES 22
+#else
+#define MAX_DESC_ENTRIES 16
+#endif
+
+	CIPHER_TRACE("Allocate Algo 0x%" PRIX32 " Context (%p)", algo, ctx);
+
+	alg = get_cipheralgo(algo);
+	if (!alg) {
+		CIPHER_TRACE("Algorithm not supported");
+		return TEE_ERROR_NOT_IMPLEMENTED;
+	}
+
+	cipherdata = caam_calloc(sizeof(*cipherdata));
+	if (!cipherdata) {
+		CIPHER_TRACE("Allocation Cipher data error");
+		return TEE_ERROR_OUT_OF_MEMORY;
+	}
+
+	/* Allocate the descriptor */
+	cipherdata->descriptor = caam_calloc_desc(MAX_DESC_ENTRIES);
+	if (!cipherdata->descriptor) {
+		CIPHER_TRACE("Allocation descriptor error");
+		ret = TEE_ERROR_OUT_OF_MEMORY;
+		goto err_allocate;
+	}
+
+	/* Setup the Algorithm pointer */
+	cipherdata->alg = alg;
+
+	/* Initialize the block buffer */
+	cipherdata->blockbuf.max = cipherdata->alg->size_block;
+
+	*ctx = cipherdata;
+
+	return TEE_SUCCESS;
+
+err_allocate:
+	caam_free_desc(&cipherdata->descriptor);
+	caam_free(cipherdata);
+
+	return ret;
+}
+
+/*
+ * Free the internal cipher data context
+ *
+ * @ctx    Caller context variable
+ */
+static void do_free_intern(struct cipherdata *ctx)
+{
+	CIPHER_TRACE("Free Context (%p)", ctx);
+
+	if (ctx) {
+		/* Free the descriptor */
+		caam_free_desc(&ctx->descriptor);
+
+		/* Free the Key 1  */
+		caam_free_buf(&ctx->key1);
+
+		/* Free the Key 2  */
+		caam_free_buf(&ctx->key2);
+
+		/* Free the Tweak */
+		caam_free_buf(&ctx->tweak);
+
+		/* Free the Context Register */
+		caam_free_buf(&ctx->ctx);
+
+		/* Free Temporary buffer */
+		caam_free_buf(&ctx->blockbuf.buf);
+	}
+}
+
+/*
+ * Free the SW Cipher data context
+ *
+ * @ctx    Caller context variable
+ */
+static void do_free(void *ctx)
+{
+	CIPHER_TRACE("Free Context (%p)", ctx);
+
+	if (ctx) {
+		do_free_intern(ctx);
+		caam_free(ctx);
+	}
+}
+
+/*
+ * Copy Software Cipher Context
+ *
+ * @dst_ctx  [out] Reference the context destination
+ * @src_ctx  Reference the context source
+ */
+static void do_copy_state(void *dst_ctx, void *src_ctx)
+{
+	struct cipherdata *dst = dst_ctx;
+	struct cipherdata *src = src_ctx;
+
+	CIPHER_TRACE("Copy State context (%p) to (%p)", src_ctx, dst_ctx);
+
+	dst->alg = src->alg;
+	dst->encrypt = src->encrypt;
+
+	if (src->blockbuf.filled) {
+		struct caambuf srcdata = {
+			.data = src->blockbuf.buf.data,
+			.length = src->blockbuf.filled };
```

Comment:
I'd prefer having the closing `}` on next line instead. Same for all the other places here.

Reply:
Ok